lib/Dialect/OM/Transforms/LinkModules.cpp

```diff
@@ -55,16 +63,29 @@ struct LinkModulesPass : public LinkModulesBase<LinkModulesPass> {
 
 LogicalResult ModuleInfo::initialize() {
   for (auto &op : llvm::make_early_inc_range(module.getOps())) {
-    if (auto classLike = dyn_cast<ClassLike>(op))
-      symbolToClasses.insert({classLike.getSymNameAttr(), classLike});
-    else
-      op.erase();
+    // If the op delares a symbol.
+    if (op.hasAttr(symAttrName)) {
```

Comment:
I think this makes sense, but it might be a little more generic to query if the op has the Symbol op interface. I.e., something like:

```
if (auto symbol = dyn_cast<SymbolOpInterface>(op))
  auto sym = op.getNameAttr();
  ...
```

Reply:
Makes sense, updated.